engine/src/main/java/com/google/android/fhir/db/Database.kt

```diff
@@ -105,6 +105,12 @@ internal interface Database {
    */
   suspend fun getAllLocalChanges(): List<LocalChange>
 
+  /**
+   * Retrieves all [LocalChange]s for the [Resource] which has the [LocalChange] with the minimum
+   * [LocalChange.timestamp]
+   */
+  suspend fun getAllChangesForResourceWithEarliestChange(): List<LocalChange>
```

Comment:
What do you think of this name? imo, it is cleaner
```suggestion
  suspend fun getAllChangesForEarliestChangedResource(): List<LocalChange>
```